compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs

```diff
@@ -208,25 +209,25 @@ impl<'tcx> ConstToPat<'tcx> {
                 let field = FieldIdx::new(idx);
                 // Patterns can only use monomorphic types.
                 let ty = self.tcx.normalize_erasing_regions(self.typing_env, ty);
-                FieldPat { field, pattern: self.valtree_to_pat(val, ty) }
+                let cv = ty::Value { valtree: val, ty };
+                FieldPat { field, pattern: self.valtree_to_pat(cv) }
             })
             .collect()
     }
 
     // Recursive helper for `to_pat`; invoke that (instead of calling this directly).
-    // FIXME(valtrees): Accept `ty::Value` instead of `Ty` and `ty::ValTree` separately.
     #[instrument(skip(self), level = "debug")]
-    fn valtree_to_pat(&self, cv: ValTree<'tcx>, ty: Ty<'tcx>) -> Box<Pat<'tcx>> {
+    fn valtree_to_pat(&self, cv: ty::Value<'tcx>) -> Box<Pat<'tcx>> {
```

Comment:
ok yea in hindsight I think this is actually not an improvement and I don't know how to improve it. Thoughts @RalfJung ?

Reply:
If it's not an improvement then let's not do it. :shrug: I can't dig into the code right now to suggest a refactor, sorry.

Reply:
Agreed

@FedericoBruzzone please remove the const_to_pat changes and the FIXME. let's stick with what we have for now

Reply:
Doing so will leave `const_to_pat` as one of the few (if not the only fn) that accepts them separately. Shall I proceed?

Reply:
yes